File: 3024-type-of-triangle.cpp
/*
3024. Type of Triangle

Submitted: October 29, 2024

Runtime: 0 ms (beats 100.00%)
Memory: 22.50 MB (beats 73.04%)
*/

class Solution {
public:
    string triangleType(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        return       nums[0] + nums[1] <= nums[2] ? "none" : 
         nums[0] == nums[1] && nums[1] == nums[2] ? "equilateral" : 
         nums[0] == nums[1] || nums[1] == nums[2] ? "isosceles" : 
                                                    "scalene";
    }
};

File: 3024-type-of-triangle.py
"""
3024. Type of Triangle

Submitted: June 18, 2025

Runtime: 0 ms (beats 100.00%)
Memory: 17.51 MB (beats 96.23%)
"""

equilateral = "equilateral"
isosceles = "isosceles"
scalene = "scalene"
none = "none"
class Solution:
    def triangleType(self, nums: List[int]) -> str:
        nums.sort()
        if nums[0] == nums[1] == nums[2]: return equilateral
        if nums[0] + nums[1] <= nums[2]: return none
        if nums[0] == nums[1] or nums[1] == nums[2]: return isosceles
        return scalene
